packages/core/src/data-grid/data-grid-lib.ts

```diff
@@ -307,11 +307,17 @@ export function drawTextCell(args: BaseDrawArgs, data: string) {
     }
 
     if (data.length > 0) {
-        const dir = direction(data);
+        if (theme.textAlign) {
+            ctx.textAlign = theme.textAlign;
+        } else if (direction(data) === "rtl") {
+            // Set text align to end as default for RTL text
+            ctx.textAlign = "end";
+        }
 
-        if (dir === "rtl") {
-            const textWidth = measureTextCached(data, ctx, `${theme.baseFontStyle} ${theme.fontFamily}`).width;
-            ctx.fillText(data, x + w - theme.cellHorizontalPadding - textWidth + 0.5, y + h / 2);
+        if (ctx.textAlign === "right" || ctx.textAlign === "end") {
+            ctx.fillText(data, x + w - (theme.cellHorizontalPadding + 0.5), y + h / 2);
+        } else if (ctx.textAlign === "center") {
```

Comment:
Reading out of the state machine like this is super expensive and round trips in and out of JS. Cache this value.

Reply:
Good to know, I will change this.

Reply:
The expensive call here is checking the property state in `ctx`, or? So, checking the value in `theme` or cell props would not be expensive?

Reply:
Correct. Reading out of ctx is expensive if done frequently (this is frequent). Setting anything in ctx can often be orders of magnitude more expensive than you might expect. Changing the text alignment as very very not cheap and thus should be done as minimally as possible.